src/otx/algo/detection/rtdetr.py

```diff
@@ -0,0 +1,438 @@
+            feat_strides=[8, 16, 32],
+            hidden_dim=256,
+            expansion=0.5,
+            dim_feedforward=1024,
+            eval_spatial_size=self.image_size[2:],
+        )
+        decoder = RTDETRTransformer(
+            num_classes=num_classes,
+            num_decoder_layers=3,
+            feat_channels=[256, 256, 256],
+            feat_strides=[8, 16, 32],
+            hidden_dim=256,
+            eval_spatial_size=self.image_size[2:],
+        )
+        criterion = RTDetrCriterion(
+            weight_dict={"loss_vfl": 1.0, "loss_bbox": 5, "loss_giou": 2},
+            num_classes=num_classes,
+            gamma=2.0,
+            alpha=0.75,
+        )
+        optimizer_configuration = [
+            {"params": "^(?=.*backbone)(?=.*norm).*$", "weight_decay": 0.0, "lr": 0.00001},
+            {"params": "^(?=.*backbone)(?!.*norm).*$", "lr": 0.00001},
+            {"params": "^(?=.*(?:encoder|decoder))(?=.*(?:norm|bias)).*$", "weight_decay": 0.0},
+        ]
+
+        return RTDETR(
+            backbone=backbone,
+            encoder=encoder,
+            decoder=decoder,
+            num_classes=num_classes,
+            criterion=criterion,
+            optimizer_configuration=optimizer_configuration,
+        )
+
+
+class OTXRtDetr50(OTXRtDetr):
+    load_from = (
+        "https://github.com/lyuwenyu/storage/releases/download/v0.1/rtdetr_r50vd_2x_coco_objects365_from_paddle.pth"
+    )
+
+    def _build_model(self, num_classes: int) -> nn.Module:
+        backbone = PResNet(
+            depth=50,
+            return_idx=[1, 2, 3],
+            num_stages=4,
+            pretrained=True,
+            freeze_at=0,
+            norm_cfg={"type": "FBN", "name": "norm"},
+        )
+        encoder = HybridEncoder(
+            in_channels=[512, 1024, 2048],
+            feat_strides=[8, 16, 32],
+            hidden_dim=256,
+            expansion=1.0,
+            dim_feedforward=1024,
+            eval_spatial_size=self.image_size[2:],
+        )
+        decoder = RTDETRTransformer(
+            num_classes=num_classes,
+            feat_channels=[256, 256, 256],
+            feat_strides=[8, 16, 32],
+            hidden_dim=256,
+            num_levels=3,
+            num_queries=300,
+            eval_spatial_size=self.image_size[2:],
+            num_decoder_layers=6,
+            num_denoising=100,
+            eval_idx=-1,
+        )
+
+        optimizer_configuration = [
+            {"params": "backbone", "lr": 0.00001},
+            {"params": "^(?=.*decoder(?=.*bias|.*norm.*weight)).*$", "weight_decay": 0.0},
+            {"params": "^(?=.*encoder(?=.*bias|.*norm.*weight)).*$", "weight_decay": 0.0},
+        ]
+
+        return RTDETR(
+            backbone=backbone,
+            encoder=encoder,
+            decoder=decoder,
+            num_classes=num_classes,
+            optimizer_configuration=optimizer_configuration,
+        )
+
+
+class OTXRtDetr101(OTXRtDetr):
```

Comment:
I'd like to suggest renaming it to `RTDETRRes101` referring to the official repo.
https://github.com/lyuwenyu/RT-DETR?tab=readme-ov-file#-implementations

Reply:
I ended up with RTDETR101
https://github.com/openvinotoolkit/training_extensions/pull/3741/commits/2d5cef1ef2bbc576bebd10fd1a96e812d6ef0548